src/paperoni/collection/filecoll.py

```diff
@@ -1,66 +1,47 @@
-import json
-from dataclasses import dataclass
-from functools import cached_property
+from dataclasses import dataclass, field
 from pathlib import Path
-from typing import Iterable
 
-from serieux import deserialize, serialize
+from serieux import deserialize, dump, load, serialize
 
-from ..model.classes import Paper
-from .tmpcoll import TmpCollection
-
-_id_types = {
-    "arxiv",
-    "dblp",
-    "doi",
-    "mlr",
-    "openalex",
-    "openreview",
-    "pmc",
-    "pubmed",
-    "semantic_scholar",
-    "uid",
-}
+from ..model.classes import CollectionPaper
+from .memcoll import MemCollection
 
 
 @dataclass
-class FileCollection(TmpCollection):
-    directory: Path
+class FileCollection(MemCollection):
+    directory: Path = field(compare=False)
+    _last_id: int = field(init=False, compare=False, default=None)
+    _papers: list[CollectionPaper] = field(init=False, default_factory=list)
+    _exclusions: set[str] = field(init=False, default_factory=set)
 
     def __post_init__(self):
-        if not self.directory.exists():
+        if self.directory is None:
+            # If no directory is provided, use the MemCollection implementation
+            self._file = None
+
+            super().__post_init__()
+            return
+
+        self._file = self.directory / "collection.yaml"
```

Comment:
Since we're storing everything in one file, I think we should pass the file, not a directory. Also that file should not be None.

Reply:
fixed